### ISSUE TITLE
RuntimeError: dictionary changed size during iteration

### ISSUE THREAD

**Issue by shner-elmo** [langchain-ai/langgraph#2119]
### Checked other resources

- [X] I added a very descriptive title to this issue.
- [X] I searched the [LangGraph](https://langchain-ai.github.io/langgraph/)/LangChain documentation with the integrated search.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I am sure that this is a bug in LangGraph/LangChain rather than my code.
- [X] I am sure this is better as an issue [rather than a GitHub discussion](https://github.com/langchain-ai/langgraph/discussions/new/choose), since this is a LangGraph bug and not a design question.

### Example Code

```python
from app.graph.graph import build_graph

graph = build_graph(...)
events = graph.stream({...}, stream_mode='values')

events_filtered = []
for index, event in enumerate(events):
    if event.get(''):        
        events_filtered.append(event)
```

### Error Message and Stack Trace (if applicable)

```shell
RuntimeError: dictionary changed size during iteration
  File "celery/app/trace.py", line 736, in __protected_call__
    return self.run(*args, **kwargs)
  File "app/main.py", line 114, in process_message_task
    return asyncio.run(process_message(my_dict))
  File "asyncio/runners.py", line 190, in run
    return runner.run(main)
  File "asyncio/runners.py", line 118, in run
    return self._loop.run_until_complete(task)
  File "asyncio/base_events.py", line 654, in run_until_complete
    return future.result()
  File "app/main.py", line 270, in process_message
    for index, event in enumerate(events):
  File "/opt/render/project/src/.venv/lib/python3.11/site-packages/langgraph/pregel/__init__.py", line 920, in stream
    with SyncPregelLoop(
  File "/opt/render/project/src/.venv/lib/python3.11/site-packages/langgraph/pregel/loop.py", line 524, in __exit__
    return self.stack.__exit__(exc_type, exc_value, traceback)
  File "contextlib.py", line 601, in __exit__
    raise exc_details[1]
  File "contextlib.py", line 586, in __exit__
    if cb(*exc_details):
  File "/opt/render/project/src/.venv/lib/python3.11/site-packages/langgraph/pregel/executor.py", line 84, in __exit__
    if tasks := {t for t in self.tasks if not t.done()}:
  File "/opt/render/project/src/.venv/lib/python3.11/site-packages/langgraph/pregel/executor.py", line 84, in <setcomp>
    if tasks := {t for t in self.tasks if not t.done()}:
```

### Description

TLDR; I'm iterating over an `Iterator[dict]` (returned by `graph.stream()`), and I get the error mentioned above.
Its very odd, because on my side the only thing I'm doing with the namespace `events: Iterator[dict]` is looping over it, and with `event: dict` is only calling `dict.get`, so only reading, not mutating.

Yet I get this error of modifying while iterating, I would imagine the error comes from `langgraph`. I suppose it has something to do with `async`? perhaps its modified by another "thread" concurrently? 

### System Info

System Information
------------------
> OS:  Linux
> OS Version:  #40~22.04.3-Ubuntu SMP PREEMPT_DYNAMIC Tue Jul 30 17:30:19 UTC 2
> Python Version:  3.11.4 (main, Jun 18 2023, 17:04:26) [GCC 11.3.0]

Package Information
-------------------
> langchain_core: 0.2.27
> langchain: 0.2.12
> langchain_community: 0.2.11
> langsmith: 0.1.96
> langchain_anthropic: 0.1.22
> langchain_groq: 0.1.9
> langchain_openai: 0.1.20
> langchain_text_splitters: 0.2.2
> langgraph: 0.2.14

Packages not installed (Not Necessarily a Problem)
--------------------------------------------------
The following packages were not found:

> langserve

**Comment by nfcampos** [comment]
Thanks for reporting, opened PR to fix

**Comment by shner-elmo** [comment]
Wow that amazing! how did you find the fix so fast?

Anyways thanks
